src/main/kotlin/minesweeper/application/RandomBoardGenerator.kt

```diff
@@ -0,0 +1,33 @@
+package minesweeper.application
+
+import minesweeper.domain.Board
+import minesweeper.domain.Coordinate
+import minesweeper.domain.EmptyCell
+import minesweeper.domain.MinedCell
+
+class RandomBoardGenerator : BoardGenerator {
+    override fun generate(command: GenerateMinesweeperCommand): Board {
+        val width = command.width
+        val area = command.area
+        val mineCount = command.mineCount
+
+        val shuffledIndices = (0 until area).shuffled()
+
+        val minedCells =
+            shuffledIndices
+                .take(mineCount)
+                .map { MinedCell(toCoordinate(it, width)) }
+
+        val emptyCellCoordinates =
+            shuffledIndices
+                .drop(mineCount)
+                .map { EmptyCell(toCoordinate(it, width)) }
+
+        return Board((minedCells + emptyCellCoordinates).toSet())
+    }
+
+    private fun toCoordinate(
+        index: Int,
+        width: Int,
+    ): Coordinate = Coordinate(index / width, index % width)
```

Comment:
저는 지금 Coordinate와 같이 

- 연속된 같은 타입의 인자를 넣어줘야할 때
- 인자의 이름이 비슷한 경우 (헷갈릴 가능성이 많을때)

휴먼에러를 방지하기위해 named parameters를 명시하는 편인데요!! 지완님도 이부분을 고민해보시면 좋을 것같습니다!! 이런 컨벤션이 팀단위 코드를 작성할때 가독성도 올라가더라구요!

```suggestion
    ): Coordinate = Coordinate(y = index / width, x = index % width)
```

Reply:
전적으로 동의합니다.